mbf_costmap_nav/src/mbf_costmap_nav/free_pose_search.cpp

```diff
@@ -329,7 +346,16 @@ SearchSolution FreePoseSearch::search() const
       in_queue_or_visited.insert(cell_index);
       queue.push(neighbor);
     }
-  }
+
+    if (queue.empty())
+    {
+      break;
+    }
+
+    test_cell = queue.top();
+    queue.pop();
+    costmap2d->mapToWorld(test_cell.x, test_cell.y, sol.pose.x, sol.pose.y);
```

Comment:
R: I find this while loop to be very un-intuitive. You're using variables that were set from outside as the variables to evaluate. Can you refactor this to follow the standard flow? Something like:

```cpp
while (!queue.empty())
{
    Cell test_cell = queue.top();
    queue.pop();
    SearchSolution sol;
    costmap2d->mapToWorld(test_cell.x, test_cell.y, sol.pose.x, sol.pose.y);

    // ...
}
```

Remove `test_cell` and `sol` from outside the loop, and simply add the poses to the queue.
I guess you need to test the goal pose outside the loop, so create a new Cell object for that rather than re-using `test_cell`